src/main/java/ca/uwaterloo/drinkmasterapi/feature/mqtt/service/MqttClientServiceImpl.java

```diff
@@ -0,0 +1,135 @@
+        ObjectNode otaObject = objectMapper.createObjectNode();
+        otaObject.put("locationId", locationId);
+        otaObject.put("productIndex", productIndex);
+        otaObject.put("md5", digest);
+        otaObject.put("url", url);
+        otaObject.put("device", "tablet/label");
+        otaObject.put("build", "prod/dev");
+        AWSIotMessage pub = new AWSIotMessage(OTA_TOPIC, AWSIotQos.QOS1, otaObject.toString());
+        this.awsIotMqttClient.publish(pub);
+        logger.info("ota message is sent");
+    }
+
+    @Override
+    public void publishLabelMessage(final String locationId, final String baseUrl, final String productIndex) throws AWSIotException {
+        ObjectMapper objectMapper = new ObjectMapper();
+        ObjectNode labelObject = objectMapper.createObjectNode();
+        labelObject.put("locationId", locationId);
+        labelObject.put("productIndex", productIndex);
+        labelObject.put("baseUrl", baseUrl);
+        AWSIotMessage pub = new AWSIotMessage(LABEL_TOPIC, AWSIotQos.QOS1, labelObject.toString());
+        this.awsIotMqttClient.publish(pub);
+        logger.info("label message is sent");
+    }
+
+    @Override
+    public void publishPourMessage(Integer id, Integer machineId, Integer transId, String time, String content) throws AWSIotException {
+        ObjectMapper objectMapper = new ObjectMapper();
+        ObjectNode pourObject = objectMapper.createObjectNode();
+        pourObject.put("id", id);
+        pourObject.put("transId", transId);
+        pourObject.put("time", time);
+        pourObject.put("content", content);
+        AWSIotMessage pub = new AWSIotMessage("aws_iot/" + machineId + "/" + POUR_TOPIC, AWSIotQos.QOS1, pourObject.toString());
+        this.awsIotMqttClient.publish(pub);
+        logger.info("pour cmd is sent");
+    }
+
+    private AWSIotMqttClient initMqttClient(String clientEndpoint, String clientId, String accessKey, String secretKey, String region) throws AWSIotException {
+        AWSCredentials creds = new BasicAWSCredentials(accessKey, secretKey);
+        AWSSecurityTokenService sts = AWSSecurityTokenServiceClientBuilder.standard()
+                .withRegion(region)
+                .withCredentials(new AWSStaticCredentialsProvider(creds))
+                .build();
+        AWSIotMqttClient awsIotMqttClient = null;
+        final GetSessionTokenRequest getSessionTokenRequest = new GetSessionTokenRequest();
+        final GetSessionTokenResult sessionTokenResult = sts.getSessionToken(getSessionTokenRequest);
+        final Credentials credentials = sessionTokenResult.getCredentials();
+        awsIotMqttClient = new AWSIotMqttClient(clientEndpoint, clientId, credentials.getAccessKeyId(), credentials.getSecretAccessKey(), credentials.getSessionToken());
+        if (awsIotMqttClient == null) {
+            throw new IllegalArgumentException("Failed to construct client due to missing certificate or credentials.");
```

Comment:
throwed some exceptions here but it's not throwable for the function because it only `throws AWSIotException`
can throw just `Exception` and handle different types of exception if necessary